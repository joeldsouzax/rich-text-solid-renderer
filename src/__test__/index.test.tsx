/** @format */

import { BLOCKS, Document, MARKS } from '@contentful/rich-text-types';
import { cleanup, render } from 'solid-testing-library';
import { headingDoc, hrDoc, marksDoc, paragraphDoc } from './documents';
import SolidRichText from '../index';

describe('rich text solid renderer', () => {
  afterEach(cleanup);

  it('should give an empty div element when given an empty document', () => {
    const document: Document = {
      nodeType: BLOCKS.DOCUMENT,
      data: {},
      content: [],
    };

    const tree = render(() => (
      <SolidRichText document={document} />
    )).baseElement;

    expect(tree).toMatchSnapshot();
  });

  it('returns an tree of elements when given a populated document', () => {
    const document: Document = hrDoc;
    const tree = render(() => (
      <SolidRichText document={document} />
    )).baseElement;
    expect(tree).toMatchSnapshot();
  });

  it('renders node with default node renderers', () => {
    const docs: Document[] = [
      paragraphDoc,
      headingDoc(BLOCKS.HEADING_1),
      headingDoc(BLOCKS.HEADING_2),
    ];

    docs.forEach((doc) => {
      const tree = render(() => <SolidRichText document={doc} />).baseElement;
      expect(tree).toMatchSnapshot();
      cleanup();
    });
  });

  it('renders nodes with default mark renderer', () => {
    const docs: Document[] = [
      marksDoc(MARKS.ITALIC),
      marksDoc(MARKS.BOLD),
      marksDoc[MARKS.UNDERLINE],
      marksDoc[MARKS.CODE],
    ];

    docs.forEach((doc) => {
      const tree = render(() => <SolidRichText document={doc} />).baseElement;

      expect(tree).toMatchSnapshot();
      cleanup();
    });
  });
});
