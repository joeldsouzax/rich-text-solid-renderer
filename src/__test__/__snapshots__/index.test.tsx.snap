// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rich text solid renderer renders node with default node renderers 1`] = `
<body>
  <div>
    <p>
      hello world
    </p>
  </div>
</body>
`;

exports[`rich text solid renderer renders node with default node renderers 2`] = `
<body>
  <div>
    <h1>
      hello world
    </h1>
  </div>
</body>
`;

exports[`rich text solid renderer renders node with default node renderers 3`] = `
<body>
  <div>
    <h2>
      hello world
    </h2>
  </div>
</body>
`;

exports[`rich text solid renderer renders nodes with default mark renderer 1`] = `
<body>
  <div>
    <p>
      <i>
        hello world
      </i>
    </p>
  </div>
</body>
`;

exports[`rich text solid renderer renders nodes with default mark renderer 2`] = `
<body>
  <div>
    <p>
      <b>
        hello world
      </b>
    </p>
  </div>
</body>
`;

exports[`rich text solid renderer returns an tree of elements when given a populated document 1`] = `
<body>
  <div>
    <p>
      hello world
    </p>
    <hr />
    <p>
      
    </p>
  </div>
</body>
`;

exports[`rich text solid renderer should give an empty div element when given an empty document 1`] = `
<body>
  <div />
</body>
`;
